modules/codegen/src/smithy4s/codegen/internals/package.scala

```diff
@@ -0,0 +1,187 @@
+    object Long extends ShapeExtractor(_.asLongShape())
+    object BigDecimal extends ShapeExtractor(_.asBigDecimalShape())
+    object BigInteger extends ShapeExtractor(_.asBigIntegerShape())
+    object List extends ShapeExtractor(_.asListShape())
+    @nowarn("msg=method asSetShape in class Shape is deprecated")
+    object Set extends ShapeExtractor(_.asSetShape())
+    object Map extends ShapeExtractor(_.asMapShape())
+    object Structure extends ShapeExtractor(_.asStructureShape())
+    object Union extends ShapeExtractor(_.asUnionShape())
+    object Enumeration extends ShapeExtractor(_.asEnumShape())
+    object Service extends ShapeExtractor(_.asServiceShape())
+    object Resource extends ShapeExtractor(_.asResourceShape())
+    object Operation extends ShapeExtractor(_.asOperationShape())
+    object Member extends ShapeExtractor(_.asMemberShape())
+  }
+
+  object T {
+    object http extends TraitExtractor[HttpTrait]
+    object error extends TraitExtractor[ErrorTrait]
+    object httpError extends TraitExtractor[HttpErrorTrait]
+    object required extends TraitExtractor[RequiredTrait]
+    @annotation.nowarn("msg=class EnumTrait in package traits is deprecated")
+    object enumeration extends TraitExtractor[EnumTrait]
+    object timestampFormat extends TraitExtractor[TimestampFormatTrait]
+    object uuidFormat extends TraitExtractor[UuidFormatTrait]
+  }
+
+  object N {
+    object ObjectNode
+        extends NodeExtractor(node =>
+          node
+            .asObjectNode()
+            .asScala
+            .map(_.getMembers().asScala.map(_.leftMap(_.getValue())))
+        )
+    object MapNode
+        extends NodeExtractor(node =>
+          node
+            .asObjectNode()
+            .asScala
+            .map(_.getMembers().asScala.toList)
+        )
+    object ArrayNode
+        extends NodeExtractor(node =>
+          node.asArrayNode().asScala.map(_.asScala.toList)
+        )
+    object NumberNode
+        extends NodeExtractor(node =>
+          node.asNumberNode().asScala.map(_.getValue())
+        )
+    object NullNode {
+      def unapply(node: Node): Boolean = node.isNullNode()
+    }
+    object StringNode
+        extends NodeExtractor(node =>
+          node.asStringNode().asScala.map(_.getValue())
+        )
+    object BooleanNode
+        extends NodeExtractor(node =>
+          node.asBooleanNode().asScala.map(_.getValue())
+        )
+  }
+
+  implicit class IterOps[A](i: Iterable[A]) {
+    def filteredForeach(pf: PartialFunction[A, Unit]) = i.foreach { a =>
+      if (pf.isDefinedAt(a)) pf(a)
+      else ()
+    }
+  }
+
+  implicit class OptionalOps[A](opt: ju.Optional[A]) {
+    def asScala: Option[A] = if (opt.isPresent()) Some(opt.get()) else None
+  }
+
+  def uncapitalise(s: String) = {
+    if (s == s.toUpperCase()) s.toLowerCase()
+    else {
+      val count: Int = s.segmentLength(c => c.==('_'), 0)
+      val after = count + 1
+      s.take(after).toLowerCase + s.drop(after)
+    }
+  }
```

Comment:
Couldn't these be package private as well?

Reply:
And same with the objects (S, T, N) above?

Reply:
Awesome, sounds great 👍

Reply:
Addressed in [f61403e](https://github.com/disneystreaming/smithy4s/pull/648/commits/f61403e42de2457cfa88fdd8c30b72e2b2c48b25)